### PR TITLE
Added used tables in assessment dashboard

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Adds a widget in the assessment dashboard for displaying used tables

### Linked issues
None

### Functionality
- [x] added some UI

### Tests
- [x] verified on staging environment:

https://github.com/user-attachments/assets/64de0164-26dd-4470-b51d-b564266c8604

### CODE REVIEW

File: src/databricks/labs/ucx/queries/assessment/main/36_0_direct_filesystem_access_problems.md [hunk 1]
```diff
@@ -9,6 +9,6 @@ As a reminder, `dbfs:/` is not supported in Unity Catalog, and more generally di
 Rather, data should be accessed via Unity tables.
 
 Each row:
-- Points to a direct filesystem access detected in the code using the code path, query or workflow & task reference and start/end line & column;
+- Points to a direct filesystem access detected in the code using the code path, query or workflow & task reference;
```
Comment (ericvergnaud) [inline src/databricks/labs/ucx/queries/assessment/main/36_0_direct_filesystem_access_problems.md:12]: Fix typo, not worth a dedicated PR

File: src/databricks/labs/ucx/source_code/queries.py [hunk 2]
```diff
@@ -156,20 +198,34 @@ def lint_query(self, query: LegacyQuery) -> Iterable[QueryProblem]:
...
+        if query.query is None:
+            return
+        ctx = LinterContext(self._migration_index, CurrentSessionState())
+        collector = ctx.tables_collector(Language.SQL)
+        source_id = f"{dashboard_id}/{query.id}"
+        source_name = query.name or "<anonymous>"
+        source_timestamp = self._read_timestamp(query.updated_at)
+        source_lineage = [LineageAtom(object_type="QUERY", object_id=source_id, other={"name": source_name})]
+        for table in collector.collect_tables(query.query):
+            yield UsedTable(**asdict(table)).replace_source(
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/queries.py]:
```suggestion
            yield UsedTable(**asdict(table)).replace_source(
```
use dataclasses.replace instead

**Review by nfx — CHANGES_REQUESTED** [review]
produced records are incorrect - what's displayed on a demo is DFSA, e.g. 

```
format.`/path/to/file`
```

where `format` is usually `delta`, `parquet`, `csv`, `json`, `text`, `binaryFiles`, and some more things.

image

**Comment by ericvergnaud** [comment]
> produced records are incorrect - what's displayed on a demo is DFSA, e.g.
> 
> ```
> format.`/path/to/file`
> ```
> 
> where `format` is usually `delta`, `parquet`, `csv`, `json`, `text`, `binaryFiles`, and some more things.
> 
> image

Fixed by https://github.com/databrickslabs/ucx/pull/2841

File: src/databricks/labs/ucx/source_code/queries.py (hunk 2 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/queries.py]: simplified - replace_source uses dataclesses.replace

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**